zeroex/orderwatch/order_watcher.go

```diff
@@ -940,8 +870,48 @@ func (w *Watcher) add(orderInfos []*ordervalidator.AcceptedOrderInfo, validation
 		}
 	}
 
-	// TODO(albrow): How to handle the edge case of orders that were not
-	// added due to the max expiration time changing?
+	// HACK(albrow): We need to handle orders in the orderInfos argument that
+	// were never added due to the max expiration time effectively changing
+	// within the database transaction above. In other words, new orders that
+	// _were_ added can change the effective max expiration time, meaning some
+	// orders in orderInfos were actually not added. This should not happen
+	// often. For now, we respond by emitting an ADDED event (above) immediately
+	// followed by a STOPPED_WATCHING event. If this order was submitted via
+	// RPC, the RPC client will see a response that indicates the order was
+	// successfully added, and then it will look like we immediately stopped
+	// watching it. This is not too far off from what really happened but is
+	// slightly inefficient.
```

Comment:
Should these orders just be considered "rejected"?

Reply:
The case where a `pinned` order from RPC gets marked as `ADDED` and then `STOPPED_WATCHING` is what made me think of it, but I actually think it's the right approach for all orders.

Reply:
> Should these orders just be considered "rejected"?

Ideally yes. This is something I talked to @z2trillion about. It turns out that making this change is more complicated than it seems. I'd rather address it later and I created https://github.com/0xProject/0x-mesh/issues/837 to track this piece of tech debt.